modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/AssetManagerImpl.java

```diff
@@ -1611,4 +1626,45 @@ private Function<Optional<Event>, Optional<Event>> getEventUpdateFunction(Snapsh
       return Optional.of(event);
     };
   }
+
+  private Function<Optional<Event>, Optional<Event>> getEventUpdateFunctionOnlyAcl(Snapshot snapshot,
+      String orgId, User user) {
+    return (Optional<Event> eventOpt) -> {
+      MediaPackage mp = snapshot.getMediaPackage();
+      String eventId = mp.getIdentifier().toString();
+      Event event = eventOpt.orElse(new Event(eventId, orgId));
```

Comment:
I'm not sure what you want to skip. The event created here is for updating the event in the index, and since we only want to change the ACL, it only has the ACL.

Reply:
Instead of creating a new event in that case, you could return an empty Optional. That's the way these methods are supposed to work. But apparently the repopulate() method doesn't handle that case correctly. :(